arduino-ide-extension/src/node/arduino-daemon-impl.ts

```diff
@@ -156,6 +160,41 @@ export class ArduinoDaemonImpl
       '--log-format',
       'json',
     ];
+    if (debug) {
+      args.push('--debug');
+    }
```

Comment:
but in that case, we should take care about deleting the file after every start-up, otherwise the file size will grow indefinitely.

Reply:
Good idea!

I edited the commit message and the PR; my changes won't close #1056. I will let you wrap it up when you have figured out how to do the file-based logging.

Reply:
@kittaakos ok I asked @cmaglie to make a change in the arduino-cli in order to recreate the log file every time the IDE is restarted. In the meantime, I'm going to add the flag because it already works anyway